api/src/main/java/org/openmrs/module/reporting/definition/library/DefinitionLibraryFactory.java

```diff
@@ -0,0 +1,16 @@
+package org.openmrs.module.reporting.definition.library;
+
+import java.util.Collection;
+
+/**
+ * Implementing a bean of this interface allows you to dynamically create zero or more {@link DefinitionLibrary}s.
+ */
+public interface DefinitionLibraryFactory {
+
+	/**
+	 * An implementation of this interface can count on this method being invoked once by the Reporting module's activator.
+	 * @return 0 or more definition libraries you want to register
+	 */
+	Collection<DefinitionLibrary<?>> getDefinitionLibraries();
+	
+}
```

Comment:
As I said in a ticket comment, "this is probably premature optimization." I guess that since each of these libraries builds an index once and doesn't hit the disk for every lookup, the factory isn't adding anything substantial.